### PR TITLE
Run MRI zlib tests in the C ext downstream gate

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

**Comment by eregon** [comment]
@pitr-ch Could you review?

File: mx.truffleruby/mx_truffleruby.py [hunk 1]
```diff
@@ -202,9 +202,10 @@ def ruby_testdownstream_sulong(args):
     jt('test', 'cexts')
     jt('test', 'specs', ':capi')
     jt('test', 'specs', ':library_cext')
+    jt('test', 'mri', '--cext')
     jt('test', 'mri', '--openssl')
     jt('test', 'mri', '--syslog')
-    jt('test', 'mri', '--cext')
+    jt('test', 'mri', 'zlib')
```
Comment (pitr-ch) [inline mx.truffleruby/mx_truffleruby.py]: why are other options and zlib is not?

**Comment by pitr-ch** [comment]
Could we rather have only 2 categories, tests with and without sulong? And why do we keep it broken down when sulong is our dependency now.

File: mx.truffleruby/mx_truffleruby.py (hunk 1 above)
Comment (eregon) [reply mx.truffleruby/mx_truffleruby.py]: Because it's a single test file, so this is enough.
But maybe it's good to accept `--zlib` for consistency.

**Comment by eregon** [comment]
> Could we rather have only 2 categories, tests with and without sulong?

I think it might be useful to run "all openssl tests" and no other cexts, etc.
This is not so easy without the options as it's a few different files.

> And why do we keep it broken down when sulong is our dependency now.

We'd still want to only run MRI tests for C exts in the downstream gate, not all C exts tests.

Anyway, this PR is about running zlib in the downstream gate, which was not run there and a bug.
Feel free to improve further.

File: mx.truffleruby/mx_truffleruby.py (hunk 1 above)
Comment (pitr-ch) [reply mx.truffleruby/mx_truffleruby.py]: ok then

**Pull request merged**